app/src/main/java/com/example/grocerieswizard/ui/home/RecipeRecyclerViewAdapter.java

```diff
@@ -38,6 +38,10 @@ public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int
         if (viewType == VIEW_TYPE_ROW) {
             RecyclerViewRowBinding rowBinding = RecyclerViewRowBinding.inflate(LayoutInflater.from(parent.getContext()), parent, false);
             RecyclerView.ViewHolder rowHolder = new RowViewHolder(rowBinding, recipeInterface);
+
+            rowBinding.getRoot().setOnClickListener(
+                    v -> recipeInterface.onItemClick(recipeUiList.get(rowHolder.getAdapterPosition()))
+            );
             rowBinding.favIcon.setOnClickListener(v -> {
                 int position = rowHolder.getAdapterPosition();
                 if (position != RecyclerView.NO_POSITION) {
```

Comment:
Below this you have a code that adds/removes recipes from favorites, but the flow is not perfect. i will create a diagram to explain better.